File: sdk/nodejs/src/resources/proxmox_backup.js
"use strict";

const { BaseResource } = require("./base");

class ProxmoxBackupResource extends BaseResource {
  async getConnectionInfo(serviceId) {
    return this._http.get("$/api/service/{id}/pbs");
  }

  async getCredentials(serviceId) {
    return this._http.get("$/api/service/{id}/pbs/credentials");
  }

  async getUsage(serviceId) {
    return this._http.get("$/api/service/{id}/pbs/usage");
  }

  async getMetrics(serviceId) {
    return this._http.get("$/api/service/{id}/pbs/metrics");
  }

  async listSnapshots(serviceId) {
    return this._http.get("$/api/service/{id}/pbs/snapshots");
  }

  async listGroups(serviceId) {
    return this._http.get("$/api/service/{id}/pbs/groups");
  }

  async changePassword(serviceId, data) {
    return this._http.post("$/api/service/{id}/pbs/password", data);
  }

  async rotateToken(serviceId) {
    return this._http.post("$/api/service/{id}/pbs/token");
  }

  async revokeToken(serviceId) {
    return this._http.delete("$/api/service/{id}/pbs/token");
  }

}

module.exports = { ProxmoxBackupResource };

File: sdk/java/src/main/java/vn/hitechcloud/sdk/resources/ProxmoxBackup.java
package vn.hitechcloud.sdk.resources;

import java.util.Map;
import vn.hitechcloud.sdk.HttpClient;

public class ProxmoxBackup extends BaseResource {

    public ProxmoxBackup(HttpClient http) {
        super(http);
    }

    /** Get PBS info */
    public Map<String, Object> pbs(int id) throws Exception {
        return http.get(String.format("/api/service/%id/pbs", id));
    }

    /** Get credentials */
    public Map<String, Object> credentials(int id) throws Exception {
        return http.get(String.format("/api/service/%id/pbs/credentials", id));
    }

    /** Get usage */
    public Map<String, Object> usage(int id) throws Exception {
        return http.get(String.format("/api/service/%id/pbs/usage", id));
    }

    /** Get metrics */
    public Map<String, Object> metrics(int id) throws Exception {
        return http.get(String.format("/api/service/%id/pbs/metrics", id));
    }

    /** Get snapshots */
    public Map<String, Object> snapshots(int id) throws Exception {
        return http.get(String.format("/api/service/%id/pbs/snapshots", id));
    }

    /** Get groups */
    public Map<String, Object> groups(int id) throws Exception {
        return http.get(String.format("/api/service/%id/pbs/groups", id));
    }

    /** Change password */
    public Map<String, Object> changePassword(int id, Map<String, Object> data) throws Exception {
        return http.post(String.format("/api/service/%id/pbs/password", id), data);
    }

    /** Create token */
    public Map<String, Object> createToken(int id, Map<String, Object> data) throws Exception {
        return http.post(String.format("/api/service/%id/pbs/token", id), data);
    }

    /** Delete token */
    public Map<String, Object> deleteToken(int id) throws Exception {
        return http.delete(String.format("/api/service/%id/pbs/token", id));
    }

}
